frontend/ts-src/signup.ts

```diff
@@ -0,0 +1,614 @@
+/* Register a new account on Corna.
+*
+* The UI emulates a terminal based conversation with the "system".
+*/
+
+import { AxiosResponse, AxiosError } from "axios";
+
+import { request, handleNetworkError, postData } from "./lib/network";
+import {
+    createDivElement,
+    createImageElement,
+    clean,
+    isEmail,
+    handlePromise,
+    hasCapitalLetter,
+    hasDigit,
+    hasSpace,
+    spaceAtStart,
+    spaceAtEnd,
+} from "./lib/utils.js";
+
+
+enum QuestionKeys {
+    EMAIL = "email",
+    USERNAME = "username",
+    PASSWORD = "password",
+    DOMAINNAME = "domainName",
+    THEME = "theme",
+    TITLE = "title",
+    BIO = "bio",
+}
+
+
+interface RegisterConfig {
+    /* Registration config interface.
+    *
+    * Holds the state of the registration process.
+    * This is a global object
+    */
+    questionIndex: number;
+    typingSpeed: number;
+    username: SetupQuestion | undefined;
+    userInput: HTMLInputElement;
+    displayUserInput: HTMLDivElement;
+    userTyping: boolean;
+    blocked: boolean;
+    userResponses: Partial<UserResponses>;
+}
+
+
+interface ThemeOptions {
+    /* For displaying theme options during setup. */
+
+    name: string;
+    imgUrl: string;
+}
+
+
+interface SetupQuestion {
+    /* "System" conversation interface. */
+
+    key: string;
+    text: string;
+    answer?: string;
+    options?: ThemeOptions[];
+}
+
+
+interface RegisterUser {
+    username: string;
+    email: string;
+    password: string;
+}
+
+
+interface CreateCorna {
+    domainName: string;
+    title: string;
+    theme?: string;
+    bio?: string;
+}
+
+
+interface UserResponses extends CreateCorna, RegisterUser {
+    /* UserResponses to send to the server. */
+}
+
+
+const questions: SetupQuestion[] = [
+    {
+        key: "username",
+        text: "Welcome to Corna, lets get you started by picking a username",
+        answer: "",
+    },
+    {
+        key: "email",
+        text: "Please choose your email address",
+        answer: "",
+    },
+    {
+        key: "password",
+        text: "Enter an eight character password - include a capital letter and a number",
```

Comment:
Don't think your validation has anything for 8 char passwords

Reply:
Bit confused between https://github.com/MooseyAnon/corna/pull/10/commits/d3381307338db0af4217dbb1f14418973701b159#diff-f95c7caa2e231882564ef0de6a60d9205a0a5f639a9adbde0bd160552a8be25aR76 and https://github.com/MooseyAnon/corna/pull/10/commits/d3381307338db0af4217dbb1f14418973701b159#diff-843480ef208180a931162cb56636b06943ad03de6b080d6d10b9a6454ba81c37R487

Reply:
Ah, those are different files but you're right, should keep consistent naming conventions.